server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -244,7 +288,7 @@ static Mapping createDynamicUpdate(Mapping mapping, DocumentMapper docMapper, Li
         Iterator<Mapper> dynamicMapperItr = dynamicMappers.iterator();
         List<ObjectMapper> parentMappers = new ArrayList<>();
         Mapper firstUpdate = dynamicMapperItr.next();
-        parentMappers.add(createUpdate(mapping.root(), splitAndValidatePath(firstUpdate.name()), 0, firstUpdate));
+        parentMappers.add(createUpdate(mapping.root(), splitPathRespectingDisableObjects(firstUpdate.name(), docMapper), 0, firstUpdate));
```

Comment:
`splitPathRespectingDisableObjects` - this method has made the flow heavy though ? is there a way to exit faster instead of looping through finding all the objectMappers ?